Add Hive functionality to remove registered devices on removal of integration

KJonline opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

This change will remove the home assistant registered device from a users hive account, when the integration is removed within home assistant. 

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

jbouwh committed:
Cleanup and use new MQTT_BASE_SCHEMA constants (#72283)

* Use new MQTT_BASE_SCHEMA constants

* Update constants for mqtt_room and manual_mqtt

* Revert removing platform key

emontnemery committed:
Move MQTT config schemas and client to separate modules (#71995)

* Move MQTT config schemas and client to separate modules

* Update integrations depending on MQTT

bramkragten committed:
Update frontend to 20220601.0 (#72855)

bdraco committed:
Fix logbook not setting up with an recorder filter that has empty fields (#72869)

jjlawren committed:
Remove announce workaround for Sonos (#72854)

bdraco committed:
Ensure recorder shuts down when its startup future is canceled out from under it (#72866)

bdraco committed:
Fix migration of MySQL data when InnoDB is not being used (#72893)

Fixes #72883

jbouwh committed:
Update MQTT tests to use the config entry setup (#72373)

* New testframework and tests for fan platform

* Merge test_common_new to test_common

* Add alarm_control_panel

* Add binary_sensor

* Add button

* Add camera

* Add climate

* Add config_flow

* Add cover

* Add device_tracker_disovery

* Add device_trigger

* Add diagnostics

* Add discovery

* Add humidifier

* Add init

* Add lecacy_vacuum

* Add light_json

* Add light_template

* Add light

* Add lock

* Add number

* Add scene

* Add select

* Add sensor

* Add siren

* Add state_vacuum

* Add subscription

* Add switch

* Add tag

* Add trigger

* Add missed tests

* Add another missed test

* Add device_tracker

* Remove commented out code

* Correct tests according comments

* Improve mqtt_mock_entry and recover tests

* Split fixtures with and without yaml setup

* Update fixtures manual_mqtt

* Update fixtures mqtt_json

* Fix test tasmota

* Update fixture mqtt_room

* Revert fixture changes, improve test

* re-add test

matrixd2 committed:
Bump yolink-api to 0.0.6 (#72903)

* Bump yolink-api to 0.0.6

* update testcase

bdraco committed:
Only present history_stats state as unknown if the time is in the future (#72880)

allenporter committed:
Fix bug in caldav and avoid unnecessary copy of dataclass (#72922)

nojocodex committed:
Fix logging & exit code reporting to S6 on HA shutdown (#72921)

emontnemery committed:
Fix reload of MQTT yaml config (#72901)

bdraco committed:
Fix performance of logbook entity and devices queries with large MySQL databases (#72898)

Commit:
Fix Hive authentication (#72929)

balloob committed:
Only sync when HA is started up as we already sync at startup (#72940)

balloob committed:
Bumped version to 2022.6.1

bdraco committed:
Only create auto comfort entities for BAF devices that support them (#72948)

bdraco committed:
Fix misalignments between sql based filtering with the entityfilter based filtering (#72936)

balloob committed:
Merge pull request #72944 from home-assistant/rc

bdraco committed:
Remove unused code from logbook (#72950)

starkillerOG committed:
Bump pynetgear to 0.10.4 (#72965)

bump pynetgear to 0.10.4

xaf committed:
Allow `log` template function to return specified `default` on math domain error (#72960)

Fix regression for logarithm template

shbatm committed:
Check ISY994 climate for unknown humidity value on Z-Wave Thermostat (#72990)

Check ISY994 climate for unknown humidity on Z-Wave Thermostat

Update to #72670 to compare the property value and not the parent object. Should actually fix #72628

rikroe committed:
Bump bimmer_connected to 0.9.4 (#72973)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

bdraco committed:
Fix statistics_during_period being incorrectly cached (#72947)

elupus committed:
fjaraskupan: Don't filter anything in backend (#72988)

jjlawren committed:
Provide Sonos media position if duration not available (#73001)

allenporter committed:
Fix google calendar bug where expired tokens are not refreshed (#72994)

alengwenus committed:
Bump pypck to 0.7.15 (#73009)

bdraco committed:
Fix history stats not comparing all times in UTC (#73040)

bdraco committed:
Fix missing historical context data in logbook for MySQL and PostgreSQL (#73011)

balloob committed:
Bumped version to 2022.6.2

balloob committed:
Merge pull request #73041 from home-assistant/rc

bdraco committed:
Bump aiolookup to 0.1.1 (#73048)

bachya committed:
Bump regenmaschine to 2022.06.0 (#73056)

bachya committed:
Bump simplisafe-python to 2022.06.0 (#73054)

hesselonline committed:
Bump wallbox to 0.4.9 (#72978)

bachya committed:
Fix unhandled exception when RainMachine coordinator data doesn't exist (#73055)

marcelveldt committed:
Throttle multiple requests to the velux gateway (#72974)

gwww committed:
Fix elk attributes not being json serializable (#73096)

* Fix jsonifying.

* Only serialize Enums

Bikonja committed:
Remove available property from Kodi (#73103)

lymanepp committed:
Tomorrowio utc fix (#73102)

* Discard past data using local time instead of UTC

* Tweak changes to fix tests

* Cleanup

bdraco committed:
Reduce branching in generated lambda_stmts (#73042)

bdraco committed:
Send an empty logbook response when all requested entity_ids are filtered away (#73046)

bdraco committed:
Fix incompatiblity with live logbook and google_assistant (#73063)

bdraco committed:
Mark counter domain as continuous to exclude it from logbook (#73101)

rappenze committed:
Fix fibaro cover detection (#72986)

balloob committed:
Bumped version to 2022.6.3

balloob committed:
Point iAlarm XR at PyPI fork (#73143)

bdraco committed:
Fix state_changes_during_period history query when no entities are passed (#73139)

StevenLooman committed:
Bump async-upnp-client==0.31.1 (#73135)

Co-authored-by: J. Nick Koston <nick@koston.org>

balloob committed:
Merge pull request #73136 from home-assistant/rc

jjlawren committed:
Fix errors when unjoining multiple Sonos devices simultaneously (#73133)

jesserockz committed:
Fix KeyError from ESPHome media players on startup (#73149)

Commit:
Bump version of pyunifiprotect to 3.9.0 (#73168)

Co-authored-by: J. Nick Koston <nick@koston.org>

esev committed:
Bump pywemo to 0.9.1 (#73186)

mib1185 committed:
Use default None for voltage property of FritzDevice in Fritz!Smarthome (#73141)

use default None for device.voltage

mib1185 committed:
Fix creating unique IDs for WiFi switches in Fritz!Tools (#73183)

bachya committed:
Fix bugs with RainMachine zone run time sensors (#73179)

bdraco committed:
Remove sqlalchemy lambda_stmt usage from history, logbook, and statistics (#73191)

balloob committed:
Bumped version to 2022.6.4

balloob committed:
Merge pull request #73193 from home-assistant/rc

matrixd2 committed:
Bump yolink-api to 0.0.8 (#73173)

* update api libray fix hearbeat message valiation

* update yolink-api ignore invalidate message

starkillerOG committed:
Ensure netgear devices are tracked with one enabled config entry (#72969)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

d0nni3q84 committed:
Fix Feedreader Atom feeds using `updated` date (#73208)

* Feedreader: Properly support Atom feeds that use only the `updated` date format and resolve #73207.

* Revert "Feedreader: Properly support Atom feeds that use only the `updated` date format and resolve #73207."

This reverts commit 4dbd11ee04b4e8f935a22dfb51405b7bdaaba676.

* Properly support Atom feeds that use only the `updated` date format and resolve #73207.

* Revert "Properly support Atom feeds that use only the `updated` date format and resolve #73207."

This reverts commit 14366c6a2491584282b8bb96fe3779fd41849897.

* Properly support Atom feeds that use only the `updated` date format and resolve #73207.

bachya committed:
Bump regenmaschine to 2022.06.1 (#73250)

Commit:
Bumps version of pyunifiprotect to 3.9.1 (#73252)

mib1185 committed:
Fix handling of connection error during Synology DSM setup (#73248)

* dont reload on conection error during setup

* also fetch API errors during update

starkillerOG committed:
Improve Netgear logging (#73274)

* improve logging

* fix black

* invert checks

balloob committed:
Fix reloading themes crashing if no themes configured (#73287)

Commit:
Bumps version of pyunifiprotect to 3.9.2 to fix compat with protect 2.1.1 (#73299)

Commit:
Hive auth fix for users (#73247)

dullage committed:
Fix polling frequency for Starling integration (#73282)

emontnemery committed:
Fix initial tilt value of MQTT cover (#73308)

emontnemery committed:
Improve MQTT reload performance (#73313)

* Improve MQTT reload performance

* Update homeassistant/components/mqtt/__init__.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/mqtt/mixins.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/mqtt/__init__.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

hesselonline committed:
Fix wallbox sensor rounding (#73310)

balloob committed:
Guard MySQL size calculation returning None (#73331)

balloob committed:
Bumped version to 2022.6.5

balloob committed:
Merge pull request #73334 from home-assistant/rc

bdraco committed:
Filter out forced updates in live logbook when the state has not changed (#73335)

raman325 committed:
Fix zwave_js add node schemas (#73343)

* Fix zwave_js add node schemas

* Code cleanup

* Add test

Commit:
Hive Bump pyhiveapi to 0.5.10 for credentials fix (#73365)

marciogranzotto committed:
Fix smart by bond detection with v3 firmware (#73414)

bdraco committed:
Only update unifiprotect ips from discovery when the console is offline (#73411)

bdraco committed:
Fix reload race in yeelight when updating the ip address (#73390)

balloob committed:
Bump aiohue to 4.4.2 (#73420)

tetienne committed:
Fix max_value access for number platform in Overkiz (#73479)

Fix wrong property name

allenporter committed:
Fix fan support in nest, removing FAN_ONLY which isn't supported (#73422)

* Fix fan support in nest, removing FAN_ONLY which isn't supported

* Revert change to make supported features dynamic

balloob committed:
Guard withings accessing hass.data without it being set (#73454)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
Bumped version to 2022.6.6

balloob committed:
Fix unifiprotect import

balloob committed:
Merge pull request #73504 from home-assistant/rc

jfroy committed:
Bump aiobafi6 to 0.6.0 to fix logging performance (#73517)

bdraco committed:
Handle offline generators in oncue (#73568)

Fixes #73565

muppet3000 committed:
Bump growattServer to 1.2.2 (#73561)

Fix #71577 - Updating growattServer dependency

bieniu committed:
Use IP address instead of hostname in Brother integration (#73556)

mib1185 committed:
Fix voltage and current values for Fritz!DECT smart plugs (#73608)

fix voltage and current values

StevenLooman committed:
Don't verify ssl certificates for ssdp/upnp devices (#73647)

bdraco committed:
Retry on SenseAPIException during sense config entry setup (#73651)

bdraco committed:
Fix calling permanent off with nexia (#73623)

* Fix calling permanent off with nexia

Changelog: https://github.com/bdraco/nexia/compare/1.0.1...1.0.2

Fixes #73610

* one more

jbouwh committed:
Fix MQTT config schema to ensure correct validation (#73619)

* Ensure config schema validation

* Use correct schema for device_tracker

* Remove schema validation from the platform setup

* Remove loop to build schema

emontnemery committed:
Fix handling of illegal dates in onvif sensor (#73600)

* Fix handling of illegal dates in onvif sensor

* Address review comment

* Address review comment

maxgashkov committed:
Fix AmbiClimate services definition (#73668)

emontnemery committed:
Don't attempt to reload MQTT device tracker (#73577)

Kane610 committed:
Fix CSRF token for UniFi (#73716)

Bump aiounifi to v32

micha91 committed:
Update aiomusiccast (#73694)

gordallott committed:
Ensure metoffice daily are returned once daily (#72440)

* ensure metoffice daily are returned once daily

* Fixes metoffice tests for MODE_DAILY

frenck committed:
Fix Plugwise migration error (#73812)

teharris1 committed:
Insteon bug fixes (#73791)

bergdahl committed:
Fix thumbnail issues in Twitch integration (#72564)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Franck Nijhof <git@frenck.dev>

frenck committed:
Fix MQTT tests for RC

frenck committed:
Bumped version to 2022.6.7

frenck committed:
Merge pull request #73821 from home-assistant/rc

Commit:
Add functionality to remove registered devices

Commit:
Add functionality to remove registered devices

Commit:
Adding commit to force tests to run again

Commit:
Adding commit to force tests to run again

Commit:
Adding commit to force tests to run again

Commit:
Revert "Adding commit to force tests to run again"

This reverts commit dd3f25fec072822c1729d095cd987b9976eba0b1.

Commit:
Revert "Adding commit to force tests to run again"

This reverts commit dd3f25fec072822c1729d095cd987b9976eba0b1.

Commit:
Revert "Adding commit to force tests to run again"

This reverts commit dd3f25fec072822c1729d095cd987b9976eba0b1.

KJonline commented:
@MartinHjelmare As you have reviewed a lot of the Hive PR previously. I was hoping you could help on this one, I'm not sure why the tests are failing. I run them locally and they pass without issues, but the GitHub action keeps failing them.

Any thoughts?

Thanks in advance.

Khole

MartinHjelmare commented:
We're having troubles with our CI at the moment. Don't worry about it. We're working on fixing it. We may ask you to rebase once it's fixed.

GitHub-Action committed:
[ci skip] Translation update

akloeckner committed:
Add this variable to trigger-based templates (#72437)

add this variables to trigger-based templates

follow-up for https://github.com/home-assistant/core/issues/70359

ludeeus committed:
Bump awesomeversion from 22.5.2 to 22.6.0 (#74030)

MatthewFlamm committed:
Use built in unit handling for nws weather (#73981)

use built in unit handling for nws

epenet committed:
Add base Entity to pylint checks (#73902)

* Add base entity properties

* Add special case of Mapping[xxx, Any]

* Add Mapping tests

* Add entity functions

* Adjust docstring

* Add update/async_update

RenierM26 committed:
Switchbot bump Dependency 0.14.0 (#74001)

* Bump requirement.

* Switchbot depenacy update, full async.

* Update tests, remove redundant config entry check.

* Update requirements_test_all.txt

* Update requirements_all.txt

* Remove asyncio lock. Not required anymore with bleak.

* Update requirements_all.txt

* Update requirements_test_all.txt

* pyswitchbot no longer uses bluepy

epenet committed:
Add CoverEntity to pylint checks (#74036)

* Add CoverEntity to pylint checks

* Avoid false positivies on device_class

* Adjust device_class handling

* Adjust device_class again using a singleton

* Adjust device_class (again)

* Simplify DEVICE_CLASS check

* Keep device_class in base class

Kane610 committed:
Fix re-login logic when UniFi integration receives a 401 (#74013)

elupus committed:
Google Assistant diagnostics and synchronization (#73574)

* Add config flow import for local google assistant
* Add diagnostic with sync response
* Add button for device sync

emontnemery committed:
Migrate tomorrowio to native_* (#74050)

emontnemery committed:
Fix misleading comments in tomorrowio (#74049)

* Fix misleading comments in tomorrowio

* Add test

Kane610 committed:
Use aiounifi v34 to utilise orjson for better performance (#74065)

Bump aiounifi to v34

AngellusMortis committed:
Automatically add newly added devices for UniFi Protect (#73879)

shbatm committed:
ISY994: Bump pyisy to 3.0.7 (#74071)

frenck committed:
Partially revert "Switch loader to use json helper (#73872)" (#74077)

epenet committed:
Cleanup update/async_update typing in Entities (#74035)

frenck commented:
@KJonline If you rebase this PR on the latest `dev`, it should resolve the CI test issue.

frenck committed:
Remove invalid unit of measurement from Glances (#73983)

ivancea committed:
Nuki opener event on ring (#72793)

* feat(nuki): add ring action timestamp attribute

* feat(nuki): add ring action state attribute

* Emit event on Nuki Opener ring

* Removed event attributes

* Use entity registry to get entity id

* Move event firing to the async update method

* Move events code outside try-except

* Black autoformat

* Added missing period to doc

* Import order

Co-authored-by: Franck Nijhof <git@frenck.dev>

epenet committed:
Adjust reauth in awair config flow (#72386)

* Adjust config-flow type hints in awair

* Improve typing of dict arguments

* Use mapping for async_step_reauth

* Add async_step_reauth_confirm

* Don't try old token

* Adjust translations

* Adjust tests

* Update tests/components/awair/test_config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/awair/test_config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/awair/test_config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/awair/test_config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/awair/test_config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/awair/test_config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/awair/test_config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

gjohansson-ST committed:
Fix wind speed SMHI (#72999)

leroyloren committed:
Visiblity fix unit km to m (#74008)

uvjustin committed:
Clean up disabling audio in stream (#74038)

emontnemery committed:
Migrate buienradar to native_* (#74059)

GitHub-Action committed:
[ci skip] Translation update

maikukun committed:
Update tesla_powerwall to 0.3.18 (#74026)

frenck committed:
Allow partial tests to take 20 minutes, use all cores (#74079)

bdraco committed:
Bump orjson to 3.7.5 (#74083)

pyos committed:
List more private and link-local IP networks  (#74064)

List more private and link-local IP networks

The IPv6 link-local network is especially important as without it
local accounts don't work on IPv6-capable networks with no IPv6 DHCP
server.

matrixd2 committed:
Fix missing leak sensor battery expose (#74084)

epenet committed:
Fix reauth step in geocaching (#74089)

epenet committed:
Adjust config-flow reauth type hints in components (#74088)

bdraco committed:
Fix devices missing in logbook when all requested entities are filtered (#74073)

epenet committed:
Fix mypy issues in zha core modules (#74028)

* Fix mypy issues in zha gateway, group and helpers

* Cleanup device

* Apply suggestion

* Raise ValueError

* Use hass.config.path

emontnemery committed:
Align code between group platforms (#74057)

epenet committed:
Fix mypy issues in zha store (#74032)

bdraco committed:
Add async_remove_config_entry_device to enphase_envoy (#74012)

bdraco committed:
Revert "Partially revert "Switch loader to use json helper (#73872)" (#74077)" (#74087)

jjlawren committed:
Fix clearing of Sonos library cache during regrouping (#74085)

Fix clearing of ZoneGroupState attribute cache

epenet committed:
Adjust type hints in component alarm methods (#74092)

* Adjust type hints in component alarm methods

* Undo related change

* Undo related change

epenet committed:
Adjust type hints in prosegur alarm (#74093)

* Adjust type hints in prosegur alarm

* Adjust hint

epenet committed:
Add AlarmControlPanelEntity to pylint checks (#74091)

epenet committed:
Fix reauth step in nest (#74090)

epenet committed:
Enforce flow-handler result type hint for step_* (#72834)

emontnemery committed:
Modify behavior of media_player groups (#74056)

epenet committed:
Cleanup async_update in smartthings cover (#74040)

emontnemery committed:
Modify behavior of lock groups (#74055)

balloob committed:
Remove entities from Alexa when disabling Alexa (#73999)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

epenet committed:
Enforce config-flow type hints for reauth step (#72830)

emontnemery committed:
Add support for unavailable and unknown to fan groups (#74054)

emontnemery committed:
Migrate environment_canada to native_* (#74048)

emontnemery committed:
Migrate zamg to native_* (#74034)

epenet committed:
Improve type hints in zha alarm (#74094)

* Improve type hints in zha alarm

* Allow None code

epenet committed:
Adjust tilt_position method in esphome cover (#74041)

epenet committed:
Use attributes in concord232 alarm (#74097)

emontnemery committed:
Add support for unavailable to cover groups (#74053)

epenet committed:
Use attributes in egardia alarm (#74098)

pvizeli committed:
Build opencv at core build pipeline (#73961)

epenet committed:
Use attributes in point alarm (#74111)

epenet committed:
Use attributes in lupusec alarm (#74109)

epenet committed:
Use attributes in ifttt alarm (#74107)

epenet committed:
Use attributes in ialarm alarm (#74099)

emontnemery committed:
Migrate climacell to native_* (#74039)

gjohansson-ST committed:
Fix depreciation period for Weather (#74106)

Fix period

epenet committed:
Improve type hints in mqtt and template alarms (#74101)

epenet committed:
Use attributes in nx584 alarm (#74105)

gjohansson-ST committed:
Native to Weather Template (#74060)

* Native to Weather template

* Add validation

epenet committed:
Remove zha from mypy ignore list (#73603)

epenet committed:
Use attributes in satel_integra alarm (#74103)

epenet committed:
Use attributes in totalconnect alarm (#74113)

pvizeli committed:
Update base image to 2022.06.2 (#74114)

gjohansson-ST committed:
Smhi minor fixes (#72606)

* Initial commit

* Tests

* From review comments

epenet committed:
Use attributes in spc alarm and binary sensor (#74120)

epenet committed:
Use attributes in ness_alarm alarm (#74121)

epenet committed:
Adjust remaining type hints in alarm properties (#74126)

bdraco committed:
Bump HAP-python to 4.5.0 (#74127)

epenet committed:
Use attributes in risco alarm (#74117)

epenet committed:
Use attributes in xiaomi_miio alarm (#74125)

TheJulianJES committed:
Fix ZHA color mode not being set correctly when changing light state (#74018)

pyos committed:
Add bool template filter and function (#74068)

Co-authored-by: Erik <erik@montnemery.com>

dmulcahey committed:
Add Aqara FP1 configuration entities to ZHA (#73027)

* add multi state input

* config entities

* remove multistate input sensor used for testing

* mypy

dmulcahey committed:
Add cluster attr data to ZHA device diagnostics (#70238)

* Add cluster attr data to ZHA device diagnostics

* add unsupported attributes and refactor

* remove await

* make parseable

bdraco committed:
Enable serialization of float subclasses with orjson (#74136)

epenet committed:
Use standard argument name in async_step_reauth (#74137)

epenet committed:
Adjust button type hints in components (#74132)

epenet committed:
Fix model in vicare device_info (#74135)

postlund committed:
Fix app browsing and local file streaming in Apple TV integration (#74112)

ludeeus committed:
Pin charset-normalizer to 2.0.12 (#74104)

starkillerOG committed:
Bump pynetgear to 0.10.6 (#74123)

emontnemery committed:
Add Entity.has_entity_name attribute (#73217)

hall committed:
Bump venstarcolortouch to 0.16 (#73038)

cereal2nd committed:
Add velbus buttons platform (#73323)

bdraco committed:
Adjust entity filters to make includes stronger than excludes (#74080)

* Adjust entity filters to make includes stronger than excludes

Fixes #59080

* adjust test for stronger entity glob includes

* sync with docs

epenet committed:
Use attributes in manual_mqtt alarm (#74124)

frenck committed:
Merge branch 'master' into dev

Commit:
Merge branch 'remove-registered-device-when-removing-the-home-assisatnt-integration' of https://github.com/Pyhass/core into remove-registered-device-when-removing-the-home-assisatnt-integration

Commit:
Merge branch 'remove-registered-device-when-removing-the-home-assisatnt-integration' of https://github.com/Pyhass/core into remove-registered-device-when-removing-the-home-assisatnt-integration

KJonline commented:
PR #74144 create to replace bad branch
